app/src/main/java/com/babylon/wallet/android/presentation/settings/debug/factors/SecurityFactorSamplesViewModel.kt

```diff
@@ -0,0 +1,250 @@
+package com.babylon.wallet.android.presentation.settings.debug.factors
+
+import android.text.format.DateUtils
+import com.babylon.wallet.android.domain.model.factors.FactorSourceCard
+import com.babylon.wallet.android.domain.model.factors.StatusMessage
+import com.babylon.wallet.android.presentation.common.StateViewModel
+import com.babylon.wallet.android.presentation.common.UiState
+import com.radixdlt.sargon.Account
+import com.radixdlt.sargon.DeviceFactorSource
+import com.radixdlt.sargon.FactorSourceKind
+import com.radixdlt.sargon.LedgerHardwareWalletFactorSource
+import com.radixdlt.sargon.Persona
+import com.radixdlt.sargon.Timestamp
+import com.radixdlt.sargon.extensions.asGeneral
+import com.radixdlt.sargon.extensions.kind
+import com.radixdlt.sargon.newAccountSampleMainnetAlice
+import com.radixdlt.sargon.newAccountSampleMainnetBob
+import com.radixdlt.sargon.newAccountSampleMainnetCarol
+import com.radixdlt.sargon.newAccountSampleStokenetNadia
+import com.radixdlt.sargon.newAccountSampleStokenetOlivia
+import com.radixdlt.sargon.newAccountSampleStokenetPaige
+import com.radixdlt.sargon.newDeviceFactorSourceSample
+import com.radixdlt.sargon.newLedgerHardwareWalletFactorSourceSample
+import com.radixdlt.sargon.newPersonaSampleMainnetBatman
+import com.radixdlt.sargon.newPersonaSampleMainnetRipley
+import com.radixdlt.sargon.newPersonaSampleMainnetSatoshi
+import com.radixdlt.sargon.newPersonaSampleStokenetConnor
+import com.radixdlt.sargon.newPersonaSampleStokenetHermione
+import com.radixdlt.sargon.newPersonaSampleStokenetLeiaSkywalker
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.collections.immutable.PersistentList
+import kotlinx.collections.immutable.persistentListOf
+import kotlinx.collections.immutable.toPersistentList
+import kotlinx.coroutines.flow.update
+import rdx.works.core.mapWhen
+import timber.log.Timber
+import javax.inject.Inject
+import kotlin.time.Duration.Companion.seconds
+
+@HiltViewModel
+class SecurityFactorSamplesViewModel @Inject constructor() : StateViewModel<SecurityFactorSamplesViewModel.State>() {
+
+    @Suppress("LongMethod")
+    override fun initialState(): State = State(
+        displayOnlyItems = persistentListOf(
+            newLedgerHardwareWalletFactorSourceSample().toCard(
```

Comment:
Used the Sargon global functions here to prevent cascading the @UsesSampleValues annotation throughout all the related components. I hope this is acceptable because this code is not meant to be used in production as it only serves as an example until we implement the actual feature.

Reply:
You can also use `@Opt-In(UsesSampleValues::class)` and use the samples defined too. Perfectly acceptable in such view model.